metabase/metabase
# Add new chart colors

Epic https://github.com/metabase/metabase/issues/22816

- Changes the way how colors are automatically assigned to chart series to use the new extended color palette. Please note that charts with less than 8 series should not be affected by this change and their colors should stay the same.
- Splits `colors.ts` into separate files. Makes sure that `d3` is imported only in `scales` module so other modules can be used in `static-viz` later.
- Tweaks how automatically generated colors are displayed in Admin

How to test preferred colors:
- New question -> `Orders` -> Sum by `Count`, Group by `Created At (month)`
- Make sure the color of the line is `brand`
- Change metric to `Average of Total`
- Make sure the color of the line has changed
- Add multiple `Count` metrics
- Make sure that the preferred color is used for the first one

How to test the extended color palette:
- New question -> `Orders` -> Sum by `Count`, Group by `Created At (month)` and `User -> State`
- Visualization -> Bar -> Display -> Stack
- Make sure it is possible to change automatically assigned colors

<img width="617" alt="Screenshot 2022-06-01 at 13 54 13" src="https://user-images.githubusercontent.com/8542534/171388633-ae593319-9fc0-4c26-9f75-58c5a3cf58b6.png">

<img width="1728" alt="Screenshot 2022-05-31 at 15 06 24" src="https://user-images.githubusercontent.com/8542534/171169276-32bab2c2-f9c0-4ec5-b9d1-d1f1b5b53844.png">
